File: Selenium/Activities/Activity5.py
# Import required packages from selenium
from selenium import webdriver
from selenium.webdriver import ActionChains
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver using WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:

    # Launch the URL and get the title
    driver.get("https://v1.training-support.net/selenium/input-events")
    home_title=driver.title
    print ("Title of the Home Page is : {}".format(home_title))

    # Build Left click action and execute it
    myActions = ActionChains(driver)
    myActions.click().perform()

    # Print the text on active side of dice
    dice = driver.find_element(By.CLASS_NAME, "active")
    diceText = dice.text
    print("Dice text, With left click : {} ".format(diceText))

    # Build double click action and execute it
    myActions.double_click().perform()

    # Print the text on active side of dice
    dice = driver.find_element(By.CLASS_NAME,"active")
    diceText = dice.text
    print("Dice text, With double click : {} ".format(diceText))

    # Build Right click action and execute it
    myActions.context_click().perform()

    # Print the text on active side of dice
    dice = driver.find_element(By.CLASS_NAME,"active")
    diceText = dice.text
    print("Dice text, With right click : {} ".format(diceText))

    # Close the Browser
    driver.close()

File: Selenium/Activities/Activity5.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity5 {
    public static void main(String[] args) {

        // Setup Firefox Driver
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();

        // Launch the URL and get the title
        driver.get("https://v1.training-support.net/selenium/input-events");
        String pageTitle= driver.getTitle();
        System.out.println(pageTitle);

        // Build Left click action and execute it
        Actions myActions = new Actions(driver);
        myActions.click()
                .build()
                .perform();

        // Print the text on active side of dice
        WebElement dice = driver.findElement(By.className("active"));
        String diceText = dice.getText();
        System.out.println("Dice text, With left click : " + diceText);

        // Build double click action and execute it
        myActions.doubleClick()
                .build()
                .perform();

        // Print the text on active side of dice
        dice = driver.findElement(By.className("active"));
        diceText = dice.getText();
        System.out.println("Dice text, With double click : " + diceText);

        // Build Right click action and execute it
        myActions.contextClick()
                .build()
                .perform();

        // Print the text on active side of dice
        dice = driver.findElement(By.className("active"));
        diceText = dice.getText();
        System.out.println("Dice text, With right click  : " + diceText);

        // Close the Browser
        driver.close();
    }
}
